src/lib/operators/aggregate_hash.cpp

```diff
@@ -34,20 +34,43 @@ template <typename ResultIds, typename Results, typename AggregateKey>
 typename Results::reference get_or_add_result(ResultIds& result_ids, Results& results, const AggregateKey& key,
                                               const RowID& row_id) {
   // Get the result id for the current key or add it to the id map
-  auto it = result_ids.find(key);
-  if (it != result_ids.end()) return results[it->second];
+  if constexpr (std::is_same_v<AggregateKey, EmptyAggregateKey>) {
+    if (results.empty()) {
+      results.emplace_back();
+      results[0].row_id = row_id;
+    }
+    return results[0];
+  } else {
+    auto it = result_ids.find(key);
+    if (it != result_ids.end()) return results[it->second];
+
+    auto result_id = results.size();
+
+    result_ids.emplace_hint(it, key, result_id);
 
-  auto result_id = results.size();
+    // If it was added to the id map, add the current row id to the result list so that we can revert the
+    // value(s) -> key mapping
+    results.emplace_back();
+    results[result_id].row_id = row_id;
 
-  result_ids.emplace_hint(it, key, result_id);
+    return results[result_id];
+  }
+}
 
-  // If it was added to the id map, add the current row id to the result list so that we can revert the
-  // value(s) -> key mapping
-  results.emplace_back();
-  results[result_id].row_id = row_id;
+template <typename AggregateKey>
+const AggregateKey& get_aggregate_key([[maybe_unused]] const KeysPerChunk<AggregateKey>& keys_per_chunk,
+                                      [[maybe_unused]] const ChunkID chunk_id,
+                                      [[maybe_unused]] const ChunkOffset chunk_offset) {
+  if constexpr (!std::is_same_v<AggregateKey, EmptyAggregateKey>) {
+    const auto& hash_keys = keys_per_chunk[chunk_id];
 
-  return results[result_id];
+    return hash_keys[chunk_offset];
+  } else {
+    static EmptyAggregateKey empty_aggregate_key;
```

Comment:
Since I will never get my head around that thing: could you add a comment explaining why you are using this static key here?